packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/transaction.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright 2017, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of cloud_firestore_platform_interface;
+
+/// An implementation of [TransactionPlatform] which uses [MethodChannel] to
+/// communication with native plugin
+class Transaction extends TransactionPlatform {
+  /// [FirebaseApp] name used for this [Transaction]
+  final String appName;
```

Comment:
It seems like a `FirebaseApp` would be more natural than `appName` here. Otherwise we're exposing an app-facing APi that isn't part of the core Firestore API.

Reply:
The `cloud_firestore/transaction.dart` doesn't expose this. This is used between the method channel class and the web class.